common/network-types/src/types.rs

```diff
@@ -318,6 +320,26 @@ impl RoutingOptions {
     }
 }
 
+/// Allows finding saved SURBs based on different criteria.
+#[derive(Clone, Debug, Copy, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum SurbMatcher {
+    /// Try to find a SURB that has the exact given [`HoprSenderId`].
+    Exact(HoprSenderId),
```

Comment:
Why would this exist as an option? I though all surbs are essentially pseudonym based.

Reply:
It's described in the PR description. We want to support also non-Session use-cases.